lib/game_runner.rb

```diff
@@ -1,37 +1,43 @@
 require_relative "grid"
 require_relative "printer"
+require_relative "player_input"
+require_relative "grid_printer"
 
 class GameRunner
-  attr_reader :stdout, :stdin, :grid, :printer
+  attr_reader :stdout, :stdin, :grid, :printer, :player_input, :grid_printer
 
   def initialize(stdout, stdin, grid = Grid.new)
     @stdout = stdout
     @stdin = stdin
     @grid = grid
-  end
-
-  def printer
-    @printer ||= Printer.new(stdout, grid)
+    @printer ||= Printer.new(stdout)
+    @grid_printer = GridPrinter.new(stdout, grid)
+    @player_input = PlayerInput.new(stdin, grid, printer)
   end
 
   def run
-    printer.print_welcome_message
-    demand_valid_coordinates
+    print_welcome_message
+    print_grid
+    get_valid_coordinate
+    mark_grid
+    print_grid
```

Comment:
This method reads really well. That is so important since this is the instruction set for the game. Someone who didn't know the code could very easily understand the app logic by reading this method.